app/hooks/use-pusher-event.ts

```diff
@@ -11,11 +12,22 @@ export function usePusherEvent<T extends unknown[] = unknown[]>(
   callback: (...args: T) => void
 ) {
   const pusher = usePusher();
+  const [channel, setChannel] = useState<Channel | null>(null);
 
-  useEffectOnce(() => {
-    if (!pusher) return;
+  useEffect(() => {
+    if (!pusher) return setChannel(null);
 
-    const channel = pusher.subscribe(channelName);
+    const c = pusher.subscribe(channelName);
+
+    setChannel(c);
+
+    return () => {
+      c?.unsubscribe();
+    };
+  }, [pusher, channelName]);
+
+  useEffect(() => {
```

Comment:
this useEffect ensures that it can adapt to changes in eventName or callback without having to resubscribe to the channel.